### ISSUE TITLE
Water conservation

### ISSUE THREAD

**Issue by MartinDix** [ACCESS-NRI/dev_coupling#33]
Monthly total water mass and equivalent sea level change from `cm3-run-18-09-2024-no-zap-no-iceberg`

![sealevel](https://github.com/user-attachments/assets/6eba238d-311f-428e-9bda-0f3d1fcfead3)

Even after accounting for snow accumulation (missing iceberg flux) there's a sea level decrease of ~ 3 mm/year.

**Comment by MartinDix** [comment]
From `cm3-run-24-09-2024-fix-wind-stress`

![sealevel](https://github.com/user-attachments/assets/97b22864-4953-4cec-b0b8-ea791f319926)

The seaice term now includes snow over sea ice
Land water includes soil water and river storage

The river storage increases over the first year, but even with this included the decrease in the first year is larger.

![sealevel2](https://github.com/user-attachments/assets/e2d7777b-1dad-4ab9-a916-3d3e2daa1b10)

**Comment by MartinDix** [comment]
River outflow from the UM matches river inflow in MOM so there's no error from the scaling by ocean fraction.

![outflow](https://github.com/user-attachments/assets/7e94603d-7b3c-4c28-92d9-a477de5bbe2b)

Rate of water loss is ~ 4e7 kg/s which is ~ 2.5% of the total river term.

UM has `l_inland=.true.` which means that the river flow which ends up in inland basins should be compensated.

**Comment by MartinDix** [comment]
UM is using ancillaries from `/g/data/tm70/mrd599/ancil/n96e_momO1_20201102`, created by suite `u-cp883`.  The river sequence file `qrparm.rivseq` puts ocean outflow points in the Black and Caspian Seas which are 100% land in the 1 degree ocean model. Therefore the river outflow to these points is lost. These points were fixed in the CM2 river sequence file.

In these plots river outflow points are shown in blue, inland basin in brown and blocked outflow points in red. Original on left and fixed on right. Changed points highlighted with thicker arrows.

![black_sea](https://github.com/user-attachments/assets/f4bbd933-a6e2-47e4-83a0-ddf5007da989)

![caspian_sea](https://github.com/user-attachments/assets/de3c1abd-f9e7-4ff9-ab23-2e62faef61d8)

Outflow to these blocked points was ~ 2.5e7 kg/s, so perhaps not the total solution.

`/g/data/tm70/mrd599/ancil/n96e_momO1_20201102/qrparm.rivseq` is now the corrected version so any new runs will pick up this change.

**Comment by kieranricardo** [comment]
@MartinDix nice catch! The new ancillary seems to generating NaN in the river outflow

**Comment by MartinDix** [comment]
Something went wrong in handling of masked points in the updated file. Corrected updated version is `/g/data/tm70/mrd599/ancil/n96e_momO1_20201102/qrparm.rivseq.20241001`.  `qrparm.rivseq` is the original file.

**Comment by kieranricardo** [comment]
@MartinDix having a look at `cm3-run-15-10-2024-revert-zapping` and `cm3-run-15-10-2024-river-seq-update`, the run off has definitely improved. But for some reason the total water budget barely improves, and the sea level change actually gets worse.

|                    | cm3-run-15-10-2024-revert-zapping | cm3-run-15-10-2024-river-seq-update | $\Delta$   |
|--------------------|-----------------------------------|-------------------------------------|------------|
| Runoff             | 114.9 mm/yr                       | 118.9 mm/yr                         | 4.0 mm/yr  |
| Sea level change   | -12.1 mm/yr                       | -16.7 mm/yr                         | -4.6 mm/yr |
| Total water change | -10.25 mm/yr                      | -9.5 mm/yr                          | 0.75 mm/yr |

**Comment by MartinDix** [comment]
The` riv-seq-update` run looks a bit better in the second year. The larger rate of decrease in the first year suggests we're still missing some sort of storage term.

![sealevel](https://github.com/user-attachments/assets/98cec8d0-5886-492f-a917-dbcafb883588)

**Comment by MartinDix** [comment]
Equivalent sea level change over the second year shows a clearer benefit from the change to the river sequence.

![sealevel](https://github.com/user-attachments/assets/329385d4-ceef-4e71-a9f7-a85b2e25071c)

However there's still something odd with the atmospheric budget. Calculating net flux to atmosphere to as `ocean evaporation + sublimation - ocean precipitation - river outflow` and total water as `soil water + snow + river water + atmospheric water` shows an imbalance.

![atm_budget](https://github.com/user-attachments/assets/66bcbccc-ef5a-41eb-8b7d-6f4d691c61f7)

This difference is equivalent to 1.7 mm/year, so close to the sea level change.

**Comment by MartinDix** [comment]
The JULES land surface scheme includes a 3 m thick groundwater layer below the usual 4 soil layers. This was missing from my calculation of the soil water content. For some reason the diagnostic variable for this is the saturation fraction rather than the actual water amount, STASH variables
```
0 |  281 |SATURATION FRAC IN DEEP LAYER 
8 |  250  | SATURATION FRAC IN DEEP LAYER   
```

 With this included the imbalance over year 2 is 1.9e6 kg/s = 0.12 mm/year.

![Image](https://github.com/user-attachments/assets/632c6323-c98e-4ea6-9667-09edeff92fdc)

The same calculation in a 40 year atmosphere only run with JULES gives a difference of 0.14 mm/year.

This is still larger than I'd expect.
